src/SimpleReservoirSimulator/Solver/Solver.cpp

```diff
@@ -0,0 +1,107 @@
+#include "Solver.h"
+
+Solver::Solver(int numberOfUnknowns) :
+    _numberOfUnknowns(numberOfUnknowns)
+{
+    Solution.resize(numberOfUnknowns);
+    SetupLinearSystem();
+}
+
+/// <summary>
+/// Setting up linear system based on the number of cells in the x-direction of the 1D reservoir (using Eigen linear algebra library for sparse matrices)
+/// </summary>
+void Solver::SetupLinearSystem()
+{
+    // For one-dimensional reservoir in single phase, number of unknowns (and #equations) is equal to number of cells in the x-direction 
+    _linearSystem = Eigen::SparseMatrix<double>(_numberOfUnknowns, _numberOfUnknowns);
+    _rightHandSide = Eigen::VectorXd::Zero(_numberOfUnknowns);
+}
+
+/// <summary>
+/// Set pressure in a cell to a constant value (assumption: linear system is empty or has only diagonal terms)
+/// </summary>
+/// <param name="constantPressure">Constant pressure in certain cell 'cellNumber'</param>
+/// <param name="cellNumber">Number of cell in which pressure has to be constant (means linear system has only a diagonal entry for this row/col)</param>
+void Solver::SetConstantPressureInCell(double constantPressure, int cellNumber)
+{
+    // Simplification 
+    _linearSystem.insert(cellNumber, cellNumber) = 1.0;
```

Comment:
Since it's a public method, you should guard inserting outside of expected memory